app/connectors/HIP/BusinessDetailsConnector.scala

```diff
@@ -0,0 +1,98 @@
+import models.connector.{ApiResponse, HipApiName, createCommonErrorParser}
+import models.domain.ApiResultT
+import models.error.DownstreamError
+import play.api.http.Status._
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse}
+import utils.{IdGenerator, Logging, TimeMachine}
+
+import java.net.URI
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+trait BusinessDetailsConnector {
+  def getBusinessDetails(businessId: Option[BusinessId], mtditid: Mtditid, nino: Nino)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext): ApiResultT[BusinessDetailsSuccessResponseSchema]
+}
+
+@Singleton
+class BusinessDetailsConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig: AppConfig, timeMachine: TimeMachine, idGenerator: IdGenerator)
+    extends BusinessDetailsConnector
+    with Logging {
+
+  private def getBusinessDetailsUrl(incomeSourceId: Option[BusinessId], mtdReference: Mtditid, nino: Nino): URI = {
+    val baseUrl = s"${appConfig.hipBaseUrl}/RESTAdapter/itsa/taxpayer/business-details"
+    val queryParams = Seq(
+      incomeSourceId.map(id => s"incomeSourceId=$id").getOrElse(""),
+      s"mtdReference=$mtdReference",
+      s"nino=$nino"
+    ).filter(_.nonEmpty).mkString("&")
+
+    new URI(s"$baseUrl?$queryParams")
+  }
+
+  private val additionalHeaders: Seq[(String, String)] = Seq(
+    "correlationid"         -> idGenerator.generateCorrelationId(),
+    "X-Message-Type"        -> "TaxpayerDisplay",
+    "X-Originating-System"  -> "MDTP",
+    "X-Receipt-Date"        -> timeMachine.now.toString,
+    "X-Regime-Type"         -> "ITSA",
+    "X-Transmitting-System" -> "HIP"
+  )
+
+  def getBusinessDetails(businessId: Option[BusinessId], mtditid: Mtditid, nino: Nino)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext): ApiResultT[BusinessDetailsSuccessResponseSchema] = {
+
+    val url: URI                             = getBusinessDetailsUrl(businessId, mtditid, nino)
+    val enrichedHeaderCarrier: HeaderCarrier = appConfig.mkMetadata(HipApiName.Api1171, url.toString).enrichedHeaderCarrier
+
+    implicit object BusinessDetailsHttpReads extends HttpReads[ApiResponse[BusinessDetailsSuccessResponseSchema]] {
+      override def read(method: String, url: String, response: HttpResponse): ApiResponse[BusinessDetailsSuccessResponseSchema] =
+        response.status match {
+          case OK =>
+            response.json
+              .validate[BusinessDetailsSuccessResponseSchema]
+              .fold[Either[DownstreamError, BusinessDetailsSuccessResponseSchema]](
+                errors => Left(createCommonErrorParser(method, url, response).reportInvalidJsonError(errors.toList)),
+                parsedModel => Right(parsedModel)
+              )
+          case BAD_REQUEST | UNAUTHORIZED | FORBIDDEN | NOT_FOUND | UNSUPPORTED_MEDIA_TYPE | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR |
```

Comment:
"Not found" is a valid use case, so it needs to be converted to a "Right" state, rather than "Left" as shown here.

We need this because the tailoring work we're doing will divert users to V&C if the API returns a 404.